### PR TITLE
Implement changeauth operations

### PR DESCRIPTION
**Pull request by puiterwijk**
This implements both ObjectChangeAuth and HierarchyChangeAuth.

Signed-off-by: Patrick Uiterwijk <patrick@puiterwijk.org>

### CODE REVIEW

File: tests/context_tests.rs [hunk 1]
```diff
@@ -1257,6 +1257,69 @@ mod test_policy_pcr {
...
+    fn test_hierarchy_change_auth() {
+        let mut context = create_ctx_with_session();
+
+        let random_digest = context.get_random(16).unwrap();
+        let new_auth = Auth::try_from(random_digest.value().to_vec()).unwrap();
+
+        // NOTE: If this test failed on your system, you are probably running it against a
+        //  real (hardware) TPM or one that is provisioned. This hierarchy is supposed to be
+        //  used by the platform. It's used in this test because if we fail at restoring auth,
+        //  it should not be a big deal on a software TPM, and it won't impact the other tests.
```
Comment (ionut-arm) [inline tests/context_tests.rs:1313]: I do hope people don't decide to run all these tests against a real TPM 😬

File: tests/context_tests.rs [hunk 2]
```diff
@@ -1257,6 +1257,69 @@ mod test_policy_pcr {
...
+        let mut context = create_ctx_with_session();
+
+        let prim_key_handle = context
+            .create_primary_key(
+                Hierarchy::Owner,
+                &decryption_key_pub(),
+                None,
+                None,
+                None,
+                PcrSelectionListBuilder::new().build(),
```
Comment (ionut-arm) [inline tests/context_tests.rs:1274]: Seeing this provided in each test for each `create..`, I'm in two minds about potentially changing this to an `Option` too, so we can just pass `None`

**Review by ionut-arm — APPROVED** [review]
Looks good!

File: tests/context_tests.rs (hunk 1 above)
Comment (puiterwijk) [reply tests/context_tests.rs:1313]: I know I do sometimes, just out of laziness! This note was here primarily for myself 😀

File: tests/context_tests.rs (hunk 2 above)
Comment (puiterwijk) [reply tests/context_tests.rs:1274]: Yep, makes sense. I can create a PR for that.

File: tests/context_tests.rs (hunk 1 above)
Comment (ionut-arm) [reply tests/context_tests.rs:1313]: Does it not affect the TPM? I'd be afraid of ending up with it ☠️

File: tests/context_tests.rs (hunk 2 above)
Comment (ionut-arm) [reply tests/context_tests.rs:1274]: No need to do that now, I'll raise an issue

File: tests/context_tests.rs (hunk 1 above)
Comment (puiterwijk) [reply tests/context_tests.rs:1313]: On my system it's been doing pretty decently. And I've found a couple of "out of resource" conditions there, that made me rework some code!
If my TPM breaks, I'll just call the vendor and get the motherboard of my system replaced  😀

Comment (puiterwijk) [reply tests/context_tests.rs:1313]: Honestly, the only things that will actually impact the TPM's lifespan are writes to the NV, and changing the hierarchy authentication. All the rest is not stored inside the TPM's persistent NVRAM.  (which is really the only part that will wear out)

File: tests/context_tests.rs (hunk 2 above)
Comment (puiterwijk) [reply tests/context_tests.rs:1274]: :+1:

**Review by Superhepper — APPROVED** [review]
Looks good to me

**Pull request merged**